packages/extension-vscode/src/quickfix-provider.ts

```diff
@@ -0,0 +1,69 @@
+import { CodeAction, CodeActionKind, CodeActionParams, Command, TextDocuments } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { getProblemNameFromDiagnostic } from './utils/problems';
+
+export class QuickFixActionProvider {
+    private documents: TextDocuments<TextDocument>;
+    private sourceName: string;
+
+    private quickFixActionsTemplates = new Map([
+        ['vscode-webhint/ignore-hint-project', (hintName: string) => {
+            return `Ignore hint '${hintName}' in this project`;
+        }],
+        ['vscode-webhint/ignore-problem-project', (hintProblem: string) => {
+            return `Ignore problem '${hintProblem}' in this project`;
+        }]
+    ]);
+
+    public constructor(documents: TextDocuments<TextDocument>, sourceName: string) {
+        this.documents = documents;
+        this.sourceName = sourceName;
+    }
+
+    public provideCodeActions(params: CodeActionParams): CodeAction[] | null {
+        const textDocument = this.documents.get(params.textDocument.uri);
+
+        if (!textDocument) {
+            return null;
+        }
+
+        const results: CodeAction[] = [];
+
+        params.context.diagnostics.forEach((currentDiagnostic) => {
+            // only respond to requests for specified source.
+            if (!currentDiagnostic.source || currentDiagnostic.source !== this.sourceName) {
+                return;
+            }
+
+            this.quickFixActionsTemplates.forEach((value, key) => {
+                // default title
+                let title = currentDiagnostic.code?.toString() || '';
+
+                // if it is a problem (not a hint) use the problem as a title instead
+                if (key.includes('-problem-')) {
+                    const problemName = getProblemNameFromDiagnostic(currentDiagnostic);
```

Comment:
Also, this does not currently work correctly for `hint-axe/*` as the problem name repeats `hint-axe/language` instead of the name of the underlying axe rule, e.g. `html-has-lang`. The corresponding edit to the `.hintrc` is also incorrect. It looks like this only works for `hint-compat-api/*` at the moment and separate support still needs to be added to disable individual axe rules.

Reply:
Probably best if we scope this further to only support ignoring specific problems for `hint-compat-api/*` for this PR and handle `hint-axe/*` in a follow-up.

Reply:
FYI, I updated this PR to scope problems to `hint-compat-api/*` and updating language to refer to them as "features" since we're really ignoring HTML/CSS features.